### PR TITLE
New184

### PR DESCRIPTION
**Pull request by ranjitjhala**
replaces https://github.com/liquid-rust/flux/pull/223

### CODE REVIEW

File: flux-middle/src/fhir.rs [hunk 1]
```diff
@@ -150,6 +150,29 @@ pub enum RefKind {
...
+            RefKind::Mut => WeakKind::Mut,
+        }
+    }
+    pub fn to_ref_kind(rk: WeakKind) -> RefKind {
+        match rk {
+            WeakKind::Mut => RefKind::Mut,
+            WeakKind::Shr => RefKind::Shr,
+            WeakKind::Arr => RefKind::Shr,
+        }
+    }
```
Comment (nilehmann) [inline flux-middle/src/fhir.rs:173]: This is not being used.

```suggestion
```

File: flux-middle/src/fhir.rs [hunk 2]
```diff
@@ -150,6 +150,29 @@ pub enum RefKind {
...
+    Arr,
+}
+
+impl WeakKind {
+    pub fn from_ref_kind(rk: RefKind) -> WeakKind {
+        match rk {
+            RefKind::Shr => WeakKind::Shr,
+            RefKind::Mut => WeakKind::Mut,
+        }
+    }
```
Comment (nilehmann) [inline flux-middle/src/fhir.rs:166]: The idiomatic way is to `impl From<RefKind> for WeaKind`

**Review by nilehmann — APPROVED** [review]
Looks good now.

File: flux-middle/src/fhir.rs (hunk 2 above)
Comment (ranjitjhala) [reply flux-middle/src/fhir.rs:166]: aha, thanks for the tip!

**Pull request merged**